doc/manual/src/contributing/hacking.md

```diff
@@ -1,77 +1,117 @@
-# Hacking
+# Hacking Nix itself
 
-This section provides some notes on how to hack on Nix. To get the
-latest version of Nix from GitHub:
+This section provides some notes on how to hack on Nix.
+
+
+### Get the latest version of Nix from GitHub
 
 ```console
 $ git clone https://github.com/NixOS/nix.git
 $ cd nix
 ```
 
-To build Nix for the current operating system/architecture use
 
-```console
-$ nix-build
-```
+### Build the Nix package
 
-or if you have a flake-enabled nix:
+**For the current operating system/architecture**:
 
 ```console
+$ nix-build
+
+$ # or for a flake-enabled nix:
 $ nix build
 ```
+This will build the `defaultPackage.${builtins.currentSystem}` attribute
+defined in the `flake.nix` file.
+
+**For another operating system/architecture**:
 
-This will build `defaultPackage` attribute defined in the `flake.nix`
-file. To build for other platforms add one of the following suffixes to
-it: aarch64-linux, i686-linux, x86\_64-darwin, x86\_64-linux. i.e.
+To build for other platforms use one of the following platform
+identifiers: `aarch64-linux`, `i686-linux`, `x86_64-darwin`, `x86_64-linux`.
+
+For example:
 
 ```console
 $ nix-build -A defaultPackage.x86_64-linux
+
+$ # or for a flake-enabled nix:
+$ nix build .#defaultPackage.x86_64-linux
 ```
 
-To build all dependencies and start a shell in which all environment
-variables are set up so that those dependencies can be found:
+
+### Open a development shell to hack on Nix
+
+To download (or build if they are not available) all dependencies and
+start a shell in which all environment variables are set up so that those
+dependencies can be found:
 
 ```console
 $ nix-shell
-```
+[dev-shell]$ # You are now in a development shell
 
-To build Nix itself in this shell:
+$ # or for a flake-enabled nix:
+$ nix develop
+[dev-shell]$ # You are now in a development shell
+```
+NOTE: Exiting the shell and re-opening it should not re-download/re-build
+anything.
 
+**You can use your prefered shell with**:
 ```console
-[nix-shell]$ ./bootstrap.sh
-[nix-shell]$ ./configure $configureFlags --prefix=$(pwd)/outputs/out
-[nix-shell]$ make -j $NIX_BUILD_CORES
+$ nix-shell --run "$SHELL"
+
+$ # or for a flake-enabled nix:
+$ nix develop -c $SHELL
 ```
 
-To install it in `$(pwd)/outputs` and test it:
+### Build Nix
+
+To build Nix itself in the development shell:
 
 ```console
-[nix-shell]$ make install
-[nix-shell]$ make installcheck -j $NIX_BUILD_CORES
-[nix-shell]$ ./outputs/out/bin/nix --version
+[dev-shell]$ ./bootstrap.sh
+[dev-shell]$ ./configure $configureFlags --prefix=$(pwd)/outputs/out
+[dev-shell]$ make -j $NIX_BUILD_CORES
+... (build log)
+[dev-shell]$ ./outputs/out/bin/nix --version
 nix (Nix) 3.0
 ```
 
-To run a functional test:
+
+### Install (in a local dir)
```

Comment:
"dir" is not a proper word, please expand.

But actually it's better to remove this section, see my comment about `make install`. When hacking on Nix in a nix-shell you should *always* use `make install` because that's really the only way to get a working/testable Nix.